examples/bq_file_load_benchmark/benchmark_tools/schema_creator.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import json
+import logging
+import os
+
+
+class SchemaCreator(object):
+    """Contains methods for creating schemas json schemas from file parameters.
+
+    Attributes:
+        schemas_dir(str): Directory where json schemas should be written to.
+        file_params(dict): Dictionary containing each file parameter and
+            its possible values.
+    """
+
+    def __init__(self, schemas_dir, file_params):
+        self.schemas_dir = schemas_dir
+        self.file_params = file_params
+
+    def create_schemas(self):
+        """Creates schemas in json format from file parameters.
+
+        Gathers combinations of schemas from the parameters in the numColumns
+        and columnTypes parameters in the self.file_params
+        dictionary. The schemas are written out in json format to the
+        self.schemas_dir directory. They can then be used to create staging
+        BigQuery tables.
+        """
+
+        # Gather file parameters.
+        num_columns = self.file_params['numColumns']
+        column_types = self.file_params['columnTypes']
+
+        # Begin iterating through parameters to create schemas.
+        for column_type in column_types:
+            # Create lists for the percents and types of columns, where each
+            # index in the percents list corresponds to the same index in the
+            # columns lists, to help with schema generation.
+            # For example, if the columnTypes param is 10_STRING_90_NUMERIC,
+            # percents will be [10, 90] and  types will be [STRING, NUMERIC].
+            percents = [float(x) for x in column_type.split('_')[::2]]
+            types = column_type.split('_')[1::2]
+            if sum(percents) != 100:
+                logging.error(('Invalid column type: {0:s}.'
+                               'Percents must equal 100.'.format(column_type)))
+            # Iterate through each value in the numColumns list from
+            # self.file_params dictionary.
+            for n in num_columns:
```

Comment:
consider refactoring these nested for loops to `itertools.product`

Reply:
I looked into it, and it wasn't working since the third loop is defined using type_count, which is defined in the second loop. Also, each run of the first loop needs exactly one `full_json` list that is used by all runs of the second loop (as opposed to one `full_json` for each run of the second loop)